src/controllers/auth.controller.js

```diff
@@ -0,0 +1,185 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { User } from '../modules/User.js';
+import { jwtService } from '../service/jwt.service.js';
+import { tokenService } from '../service/token.service.js';
+import { userService } from '../service/user.service.js';
+import bcrypt from 'bcrypt';
+import { ConsoleLoger } from '../untils/consoleLoger.js';
+
+function validateEmail(value) {
+  const EMAIL_PATTERN = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!value) {
+    return 'Email is required';
+  }
+
+  if (!EMAIL_PATTERN.test(value)) {
+    return 'Email is not valid';
+  }
+}
+
+function validatePassword(value) {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+}
+
+const register = async (req, res) => {
+  const { email, password, userName } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Email, password, are not validate', errors);
```

Comment:
The error message 'Email, password, are not validate' is grammatically incorrect. Consider changing it to 'Email or password is not valid'.